blaze-client/src/main/scala/org/http4s/client/blaze/ConnectionBuilder.scala

```diff
@@ -10,5 +10,5 @@ trait ConnectionBuilder {
   def shutdown(): Task[Unit]
 
   /** Attempt to make a new client connection */
-  def makeClient(req: Request): Task[BlazeClientStage]
+  def makeClient(req: Uri): Task[BlazeClientStage]
```

Comment:
Is there ever another case that we might need headers (maybe proxy related?) or attributes from the `Request` or will the `Uri` always uniquely determine the endpoint? I think this change is probably fine but I just wanted to raise the question.